packages/custom-config-poc-app/amplify/backend.ts

```diff
@@ -0,0 +1,219 @@
+backend.setCustomOutput('myTestMapStyle', testMapStyle, {
+  clientConfigDestinations: [
+    {
+      clientConfigFormat: 'json',
+      path: [
+        'geo',
+        'amazon_location_service',
+        'maps',
+        'items',
+        testMap.mapName,
+        'style',
+      ],
+    },
+  ],
+});
+
+const placeIndex = new PlaceIndex(locationStack, 'TestPlaceIndex');
+new iam.Policy(locationStack, 'myTestPlaceIndexPolicy', {
+  policyName: 'myTestPlaceIndexPolicy',
+  roles: [backend.auth.resources.authenticatedUserIamRole],
+  statements: [
+    new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        'geo:SearchPlaceIndexForPosition',
+        'geo:SearchPlaceIndexForText',
+        'geo:SearchPlaceIndexForSuggestions',
+        'geo:GetPlace',
+      ],
+      resources: [placeIndex.placeIndexArn],
+    }),
+  ],
+});
+
+backend.setCustomOutput('myTestSearchIndex', placeIndex.placeIndexName, {
+  clientConfigDestinations: [
+    {
+      clientConfigFormat: 'json',
+      path: ['geo', 'amazon_location_service', 'search_indices', 'default'],
+    },
+    {
+      clientConfigFormat: 'json',
+      // TODO how do we handle arrays ? Should we use json path syntax ?
+      path: ['geo', 'amazon_location_service', 'search_indices', 'items[]'],
+    },
+  ],
+});
+
+// ####### Geo End #######
+
+// ####### Alternative proposal Start #####
+
+// TODO this takes current JS form, but should really take unified config
+backend.addToClientConfig({
+  custom: {
+    myApiUrl: api.url,
+    myApiKey: apiKeyValue,
+  },
+  // TODO this is commented out because is not defined in schema.
+  // "geo": {
+  //   "amazon_location_service": {
+  //     "region": locationStack.region,
+  //     "maps": {
+  //       "default": testMap.mapName,
+  //       "items": {
+  //         `${testMap.mapName}`: {
+  //           "style": "VectorEsriNavigation"
+  //         }
+  //       }
+  //     },
+  //     "search_indices": {
+  //       "default": placeIndex.placeIndexName,
+  //       "items": [
+  //         placeIndex.placeIndexName
+  //       ]
+  //     }
+  //   }
+  // }
+});
+
+// ####### Alternative proposal End #####
+
+// ####### How big can this be Start ######
+
+// $ ll amplifyconfiguration.json
+// -rw-r--r--  1 REDACTED  1607 Jan 29 08:43 amplifyconfiguration.json
+// A size of simple config is 1607
+
+const veryLongOutput = 'a'.repeat(450000);
+new CfnOutput(locationStack, 'howbigcanoutputbe', {
+  value: veryLongOutput,
+});
+
+// Adding more outputs results in
+// Currently in UPDATE_ROLLBACK_IN_PROGRESS with reason: Outputs exceeded limit. A CloudFormation error message from handling the stack's outputs
+// So the limit seems to be around 450000 characters
+
+// client config from create-amplify template has 3342 bytes.
+
+// const veryLongOutput2 = 'a'.repeat(450000);
+// new CfnOutput(locationStack, 'howbigcanoutputbe2', {
+//   value: veryLongOutput2,
+// });
+
+// ####### How big can this be End ######
```

Comment:
Looks like all outputs combined per stack can have 450000 characters (bytes, if dumped to file size is same).
Sample client config used in this app is 1607 bytes
Client config from `create-amplify` app is 3342 bytes.

So it seem outputs can hold ~134 copies of config from `create-amplify`